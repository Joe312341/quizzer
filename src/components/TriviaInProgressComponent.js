import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, Button, TouchableOpacity, StyleSheet } from 'react-native';

const TriviaInProgressComponent = ({ currentQuestion, lastQuestion, questionProgress, answers, answeredState, onNextQuestion, onChangeAnswerStateAndScore }) => (
  <View>
    <View style={[styles.headerTile, styles.tile]}>
      <Text>Question: {questionProgress}</Text>
      <Text>Category: {currentQuestion.category}</Text>
    </View>
    <View style={[styles.questionArea, styles.tile]}>
      <Text>{currentQuestion.question}</Text>
    </View>
    <View>
      { answers.map((answer) => {
        return (
          <TouchableOpacity
            key={answer.answerText}
            disabled={answeredState}
            onPress={() => !answeredState && onChangeAnswerStateAndScore(answer.isCorrect)}
            style={answeredState && answer.isCorrect ? [styles.correctAnswerTile, styles.tile] : [styles.answerTile, styles.tile]}
            activeOpacity={0.5}
          >
            <Text>{answer.answerText}</Text>
          </TouchableOpacity>
        )
      })}
      {answeredState &&
      <Button
        onPress={() => onNextQuestion()}
        title={lastQuestion ? "Results" : "Next Question"}
        color="#841584"
      />}
    </View>
  </View>
);

const styles = StyleSheet.create({
  answerTile: {
    height: 50,
    backgroundColor: 'powderblue',
    borderColor: '#000033',
    borderWidth: 1,
  },
  correctAnswerTile: {
    height: 50,
    backgroundColor: '#85d826',
    borderColor: 'red',
    borderWidth: 1,
  },
  questionArea: {
    height: 100,
    backgroundColor: '#cddc39',
    borderColor: '#000033',
    borderWidth: 1,
  },
  headerTile: {
    backgroundColor: '#cddc39',
    borderColor: '#000033',
    borderWidth: 1,
    height: 50
  },
  tile: {
    margin: 5,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

TriviaInProgressComponent.propTypes = {
  currentQuestion: PropTypes.object.isRequired,
  lastQuestion: PropTypes.bool.isRequired,
  questionProgress: PropTypes.string.isRequired,
  answers: PropTypes.arrayOf(PropTypes.object).isRequired,
  answeredState: PropTypes.bool.isRequired,
  onNextQuestion: PropTypes.func.isRequired,
  onChangeAnswerStateAndScore: PropTypes.func.isRequired,
}

export default TriviaInProgressComponent;
