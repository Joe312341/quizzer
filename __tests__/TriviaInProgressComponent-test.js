/*global test expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TriviaInProgressComponent from '../src/components/TriviaInProgressComponent';

// question has been received by api
const currentQuestion = {
  category: "Entertainment: Cartoon & Animations",
  correct_answer: "Daffy Duck",
  difficulty: "easy",
  incorrect_answers: [], //not used
  question: "Which of these is NOT a Disney cartoon character?",
  type: "multiple"
}

const answers = [
  {
    answerText: "Scrooge McDuck",
    isCorrect: false
  },
  {
    answerText: "Daffy Duck",
    isCorrect: true
  },
  {
    answerText: "Donald Duck McDuck",
    isCorrect: false
  },
  {
    answerText: "Daisy McDuck",
    isCorrect: false
  }
]

const questionProgress = '1/2';

test('renders non answered question correctly', () => {

  const answeredState = false;
  const lastQuestion = false;
  const questionProgress = '1/2';

  const tree = renderer.create(
    <TriviaInProgressComponent
      currentQuestion={currentQuestion}
      lastQuestion={lastQuestion}
      questionProgress={questionProgress}
      answers={answers}
      answeredState={answeredState}
      onNextQuestion={()=>{}}
      onChangeAnswerStateAndScore={()=>{}}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders answered question correctly', () => {

  const answeredState = false;
  const lastQuestion = true;

  const tree = renderer.create(
    <TriviaInProgressComponent
      currentQuestion={currentQuestion}
      lastQuestion={lastQuestion}
      questionProgress={questionProgress}
      answers={answers}
      answeredState={answeredState}
      onNextQuestion={()=>{}}
      onChangeAnswerStateAndScore={()=>{}}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders answered question correctly which is the last question', () => {

  const answeredState = true;
  const lastQuestion = true;

  const tree = renderer.create(
    <TriviaInProgressComponent
      currentQuestion={currentQuestion}
      lastQuestion={lastQuestion}
      questionProgress={questionProgress}
      answers={answers}
      answeredState={answeredState}
      onNextQuestion={()=>{}}
      onChangeAnswerStateAndScore={()=>{}}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
