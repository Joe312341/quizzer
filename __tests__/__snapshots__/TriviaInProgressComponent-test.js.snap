// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders answered question correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cddc39",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Question: 
      1/2
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Category: 
      Entertainment: Cartoon & Animations
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cddc39",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 100,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Which of these is NOT a Disney cartoon character?
    </Text>
  </View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Scrooge McDuck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daffy Duck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Donald Duck McDuck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daisy McDuck
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders answered question correctly which is the last question 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cddc39",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Question: 
      1/2
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Category: 
      Entertainment: Cartoon & Animations
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cddc39",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 100,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Which of these is NOT a Disney cartoon character?
    </Text>
  </View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Scrooge McDuck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#85d826",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daffy Duck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Donald Duck McDuck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daisy McDuck
      </Text>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Results
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders non answered question correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cddc39",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Question: 
      1/2
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Category: 
      Entertainment: Cartoon & Animations
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cddc39",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 100,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Which of these is NOT a Disney cartoon character?
    </Text>
  </View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Scrooge McDuck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daffy Duck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Donald Duck McDuck
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "powderblue",
          "borderColor": "#000033",
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Daisy McDuck
      </Text>
    </View>
  </View>
</View>
`;
